### PR TITLE
fix: Return translated orgunit name in event payload [DHIS2-14596]

### PR DESCRIPTION
**Pull request by abyot**
- replace` orgunit.name` (which is untranslated version) with `orgunit.displayname` (which is translated version)
- modify SQL query to include translation values based on user locale
- existence of user locale does not mean we have always translation value, so we coalesce translated and untranslated values, and pick the first non null.

### CODE REVIEW

**Review by lucaCambi77 — COMMENTED** [review]
We need a description for a PR so it is easier to understand the changes through the code. Also, does this have test coverage? I can't see any test

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java [hunk 1]
```diff
@@ -623,7 +623,7 @@ public Event getEvent( ProgramStageInstance programStageInstance, boolean isSync
         if ( ou != null )
         {
             event.setOrgUnit( ou.getUid() );
-            event.setOrgUnitName( ou.getName() );
+            event.setOrgUnitName( ou.getDisplayName() );
         }
```
Comment (lucaCambi77) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java:627]: Why here do we add the `displayName` but later we do 
`coalesce(ou_displayname.value, ou.name) as ou_name`

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/store/TwoParamFunction.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import lombok.Getter;
+
+/**
+ * @author abyot
+ *
+ */
+@Getter
+@AllArgsConstructor
+public class TwoParamFunction implements QueryElement
+{
```
Comment (lucaCambi77) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/store/TwoParamFunction.java:40]: Do we need a @Getter here? Also,  the name `TwoParamFunction`  is not really clear to me. Either we create a function to support multiple params or we give a more meaningful name as in the case it is only about column and fallback i see

Comment (abyot) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/store/TwoParamFunction.java:40]: intention was to make it reusable for a function that takes two params. Can rename it to `TableColumnWithFallback`

**Pull request closed without merging**